crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -0,0 +1,727 @@
+use super::{events::meta_to_event_index, Metrics, Postgres};
+use anyhow::{anyhow, bail, Context, Result};
+use contracts::cowswap_onchain_orders::{
+    event_data::OrderPlacement as ContractOrderPlacement, Event as ContractEvent,
+};
+use database::{
+    byte_array::ByteArray, events::EventIndex, onchain_broadcasted_orders::OnchainOrderPlacement,
+    orders::Order, PgTransaction,
+};
+use ethcontract::{Event as EthContractEvent, H160};
+use model::{
+    order::OrderUid,
+    order::{BuyTokenDestination, OrderKind, SellTokenSource},
+    signature::SigningScheme,
+    DomainSeparator,
+    {app_id::AppId, order::OrderData},
+};
+use number_conversions::u256_to_big_decimal;
+use shared::{
+    db_order_conversions::{
+        buy_token_destination_into, order_kind_into, sell_token_source_into, signing_scheme_into,
+    },
+    event_handling::EventStoring,
+    order_quoting::{OrderQuoting, QuoteSearchParameters},
+    order_validation::{
+        convert_signing_scheme_into_quote_kind, get_quote_and_check_fee,
+        is_order_outside_market_price,
+    },
+};
+use std::collections::HashMap;
+use unzip_n::unzip_n;
+
+unzip_n!(3);
```

Comment:
Let's use https://docs.rs/itertools/latest/itertools/fn.multiunzip.html instead . We already depend on itertools in some places and it is better maintained than the unzip_n crate.